### DISCUSSION TITLE
onClear attribute for the Search component

### DISCUSSION THREAD

**Discussion by eduardo-wrkt** [mondaycom/vibe#2101] [Ideas]
Hi everyone!

I've been using the library for almost a year now and I love it, its super clean and easy to use.

One thing that I think would be super useful, is a `onClear` attribute in the **Search** component. Currently I have to add an event listener to run a function when the user clicks in the **clear button** of the Search input. With this new attribute, that wouldn't be necessary anymore.

I suppose this would be a small lift but also a major improvement for the component.

That's it. Thanks in advance.

**Comment by YossiSaadi** [comment]
Hi @eduardo-wrkt !

Thanks so much for your kind words and for being a part of our community!
It's always great to hear that our library has been useful and easy to use.

Regarding your suggestion about adding an `onClear` attribute to the `Search` component - that's a great idea!

hope you've noticed we got a new [`Search`](https://style.monday.com/?path=/docs/inputs-search--docs) component (which also does not include an `onClear` ability).
It can be imported from our `next`:
```typescript
import { Search } from "monday-ui-react-core/next";
```
We would very much like to implement your idea in our new component

We always appreciate contributions from our community members, and since you've already identified a nice enhancement, would you be interested in contributing this feature yourself? It could be a great way to get more involved with the library you like so much.

Let us know if you're interested or if you need any more information to get started!

Thanks again for your suggestion and your support!

**Reply by eduardo-wrkt** [reply]
Hey @YossiSaadi!

That would be a nice way of contributing to the library, I would love to implement this.

**Comment by eduardo-wrkt** [comment]
Hi @YossiSaadi

It took me a while, but I got back to this #2272.

I've implemented the `onClear` prop in the `Search` component as we discussed and created a PR ([#2272](https://github.com/mondaycom/vibe/pull/2272)) so the maintainers can analyse it. I hope the changes I made are useful for the library.
